### ISSUE TITLE
Setting the value of a multiline TextInput with breaks when autocorrect suggestion present

### ISSUE THREAD

**Issue by mhoran** [facebook/react-native#27019]
On iOS, when `multline={true}` is set on a `TextInput`, something wonky happens when the value of a `TextInput` is programmatically modified (via `setState` and the `value` property of said `TextInput`) and an autocorrect suggestion is present.

The value of the text input ends up being set to the combination of the autocorrect suggestion and the programmatic value. This only happens when `multiline={true}`. When `multiline={false}`, all is fine.

This seems only to be reproducible on iOS and specifically iOS 13. I was not able to reproduce on the Snack iOS simulator.

React Native version: 0.59

## Steps To Reproduce

1. Add `multiline={true}` to a `TextInput`
2. Try to programmatically change the value of the text input when an autocorrect suggestion is present.
3. The value set programmatically will become a combination of the autocorrect value and the programmatic value.

Describe what you expected to happen:

I expected the programmatic value to be set in the text input, as it is when `multiline={false}`.

Snack, code example, screenshot, or link to a repository:

https://snack.expo.io/@mhoran/6b56c3

Typing "Yo mh" and clicking the "Bananas!" text will result in the text box containing "Yo my oran", instead of "Yo mhoran". Changing `multiline={true}` to `multiline={false}` gives the expected behavior.

**Comment by mhoran** [comment]
This is still an issue.

**Comment by mhoran** [comment]
This is still an issue.

**Comment by backmeupplz** [comment]
Any news on this?

**Comment by rikur** [comment]
Related issue: https://github.com/facebook/react-native/issues/28025

**Comment by fabriziobertoglio1987** [comment]
you did not provide the info, the issue was not flagged as Platform iOS

**Comment by backmeupplz** [comment]
@fabriziobertoglio1987 what info should be provided? Sorry, I'm not OP but I can help with my case.

**Comment by rikur** [comment]
Not sure if this hack works for this case, but it could: https://github.com/facebook/react-native/issues/28025#issuecomment-631833790

**Comment by backmeupplz** [comment]
@rikur tried it before, didn't work well :( the textfield was popping in and out of editting state, not finishing at the desired state :(

**Comment by backmeupplz** [comment]
@fabriziobertoglio1987 should we do anything about this issue? Should I publish a new issue like this but with better details about the case? I mean, I have the same issue as OP.

**Comment by backmeupplz** [comment]
@Ashoat do we need to provide any more info about this?

**Comment by Ashoat** [comment]
I don't think so, the repro should be enough. I wouldn't expect anybody to solve this for you anytime soon though. If you need it fixed the best solution would be to put up a PR.

**Comment by fabriziobertoglio1987** [comment]
@backmeupplz I'm sorry. I don't work on iOS. I just work on Android. I hope somebody working on iOS will be bable to help you. Thanks a lot. Sorry I did not read the other emails. I wish you a good day. Fabrizio

**Comment by Ashoat** [comment]
The Expo snack repro doesn't appear to work in Expo 47. Would somebody mind updating it so that this issue doesn't go stale?

**Comment by mhoran** [comment]
Updated the Snack.

**Comment by mhoran** [comment]
This issue persists with React Native 0.76.0 and New Arch.

**Comment by mhoran** [comment]
This is also an issue for single-line TextInputs now as well, and and is not tied to old vs new architecture. As a workaround I was able to set `autoCorrect={false}`.

The issue does not seem to repro in the Snack with single-line TextInput but on a device I see this behavior. It could have something to do with the speed of the device.

**Comment by Bauss9** [comment]
I can't believe this issue has been active for 5 years !!!!!

TextInput value and defaultValue completely break the input's behavior, it does some weird selecting, double tap period does not work on iOS and autosuggest too ....

**Comment by mhoran** [comment]
> TextInput value and defaultValue completely break the input's behavior, it does some weird selecting, double tap period does not work on iOS and autosuggest too ....

Many issues with controlled TextInput have been fixed as of 0.76.1 so long as you are using the New Architecture. Unfortunately this issue persists.

I've updated the Snack to Expo v52 to show the continued erroneous behavior.

**Comment by mhoran** [comment]
The issue here seems to be that any interaction other than a keypress causes the autocorrect suggestion to be accepted. There is then a race condition between the autocorrect acceptance and subsequent state update via `value` property change (or direct manipulation via `setNativeProps`). Over time, this race seems to have changed from favoring the state update / direct manipulation to favoring autocorrect acceptance.

I am not how to fix this. While the expected state change does come through *after* the autocorrect suggestion is accepted, it is ignored. There *may* be a way to force *reject* an autocorrect suggestion. One could then use this when it is known that a value change may be coming (but this would be a very clunky API.) I have not found a way to intercept the autocorrect acceptance at a point when it is known that a state change is coming and that the autocorrect suggestion should be rejected.

I was able to implement another workaround using `setTimeout`, but that isn't great since the user will see a flash of incorrect text. Presumably the timeout could be tightened via native code and ensuring that the property change / direct manipulation overrides the autocorrect suggestion, but I haven't determined the feasibility of that.
